op-service/rethdb-reader/src/receipts.rs

```diff
@@ -118,8 +120,7 @@ pub(crate) unsafe fn read_receipts_inner(
     // Create a ReceiptsResult with a pointer to the json-ified receipts
     let res = ReceiptsResult::success(receipts_json.as_mut_ptr() as *mut char, receipts_json.len());
 
-    // Forget the `receipts_json` string so that its memory isn't freed by the
-    // borrow checker at the end of this scope
+    // Forget the `receipts_json` string so that its memory isn't freed by the borrow checker at the end of this scope
     std::mem::forget(receipts_json); // Prevent Rust from freeing the memory
```

Comment:
Need to also forget the new error string in `ReceiptsResult`, it will be cleaned up at the end of this scope.